gen/nvml/init.go

```diff
@@ -33,12 +33,14 @@ var nvml *dl.DynamicLibrary
 func Init() Return {
 	lib := dl.New(nvmlLibraryName, nvmlLibraryLoadFlags)
 	if lib == nil {
-		panic(fmt.Sprintf("error instantiating DynamicLibrary for %s", nvmlLibraryName))
+		log.Printf("error instantiating DynamicLibrary for %s\n", nvmlLibraryName)
+		return ERROR_UNKNOWN
```

Comment:
I don't know whether I agree with returning `ERROR_UNKNOWN` here. This mixes errors that come from outside of NVML with NVML. What is the benefit of this over, for example, a caller recovering from the panic and printing the error there instead? We could provide a wrapper that does this if required?

```
// AssertInit calls the NVML function and recovers from any panics. An error is returned if
// a panic is detected or if the return from NVML.Init is not SUCCESS.
fun AssertInit() (err error) {
    
}
```

I would like to wait for @klueska to weigh in on this though.

Reply:
The reason a panic was used here is for exactly this reason. I didn't want to use an NVML error type to denote an error in loading the library.

On a related note, there have been several requests to allow this package to be built with compile-time linkage to `libnvidia-ml.so` instead of via `dllopen()`. I will need to think about the best way to support both options and the code here will likely change as a result.